txcache/wrappedTransaction.go

```diff
@@ -1,72 +1,56 @@
 package txcache
 
 import (
-	"bytes"
-
 	"github.com/multiversx/mx-chain-core-go/data"
 )
 
-const processFeeFactor = float64(0.8) // 80%
+type BunchOfTransactions []*WrappedTransaction
 
 // WrappedTransaction contains a transaction, its hash and extra information
 type WrappedTransaction struct {
-	Tx                   data.TransactionHandler
-	TxHash               []byte
-	SenderShardID        uint32
-	ReceiverShardID      uint32
-	Size                 int64
-	TxFeeScoreNormalized uint64
-}
-
-func (wrappedTx *WrappedTransaction) sameAs(another *WrappedTransaction) bool {
-	return bytes.Equal(wrappedTx.TxHash, another.TxHash)
+	Tx              data.TransactionHandler
+	TxHash          []byte
+	SenderShardID   uint32
+	ReceiverShardID uint32
+	Size            int64
+
+	PricePerUnit uint64
+	HashFnv32    uint32
 }
 
-// estimateTxGas returns an approximation for the necessary computation units (gas units)
-func estimateTxGas(tx *WrappedTransaction) uint64 {
-	gasLimit := tx.Tx.GetGasLimit()
-	return gasLimit
-}
-
-// estimateTxFeeScore returns a normalized approximation for the cost of a transaction
-func estimateTxFeeScore(tx *WrappedTransaction, txGasHandler TxGasHandler, txFeeHelper feeHelper) uint64 {
-	moveGas, processGas := txGasHandler.SplitTxGasInCategories(tx.Tx)
-
-	normalizedMoveGas := moveGas >> txFeeHelper.gasLimitShift()
-	normalizedProcessGas := processGas >> txFeeHelper.gasLimitShift()
-
-	normalizedGasPriceMove := txGasHandler.GasPriceForMove(tx.Tx) >> txFeeHelper.gasPriceShift()
-	normalizedGasPriceProcess := normalizeGasPriceProcessing(tx, txGasHandler, txFeeHelper)
+// precomputeFields computes (and caches) the (average) price per gas unit.
+func (transaction *WrappedTransaction) precomputeFields(txGasHandler TxGasHandler) {
```

Comment:
goland weak warning because there are funcs with "transaction" and "wrappedTx" as receiver name

Reply:
Fixed - now using `wrappedTx`.

Reply:
ok